src/ralph/filters.py

```diff
@@ -1,22 +1,24 @@
 """
 Ralph tracking logs filters.
 """
+
 from .exceptions import EventKeyError
 
 
-def anonymous(events):
-    """Remove anonymous events.
+def anonymous(event):
+    """Remove anonymous event.
 
     Args:
-        events (DataFrame): events to filter
+        event (dict): event to filter
 
     Returns:
-        Filtered pandas DataFrame.
+        event (dict): when event is not anonymous
+        None: otherwise
 
     """
 
-    if events.get("username", None) is None:
-        raise EventKeyError(
-            "Cannot filter anonymous filters without 'username' column."
-        )
-    return events.loc[lambda df: df["username"] != "", :]
+    if (username := event.get("username", None)) is None:
```

Comment:
Last week I made a small poll in the team, and we decided that this pattern is hard to maintain or understand by juniors. We prefer having something more explicit/classical:

```suggestion
    username = event.get("username", None)
    if username is None:
```

Reply:
Good to know, thanks, will then avoid this pattern for now. Updated the PR.